### ISSUE TITLE
[Bug]: cant delete imported kml

### ISSUE THREAD

**Issue by mikeelucky** [nextcloud/maps#845]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Maps app **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I really need a map app. but after importing a bigger kml file i can't delete it because it's done by the url DELETE /nc/index.php/apps/maps/favorites?ids[]=2452&ids[]=2453&ids[] etc very long request and this is causes html error 414

### Steps to reproduce

1. import kml from more than +- 300 points 
2. try to remove it and nothing happend
3. watch apache access log

### Expected behavior

want to remove unused imports

### Installation method

Web installer on a VPS or web space

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
- breezedark: 23.2.2
  - bruteforcesettings: 2.4.0
  - circles: 23.0.1
  - cloud_federation_api: 1.6.0
  - contactsinteraction: 1.4.0
  - dav: 1.21.0
  - deck: 1.6.3
  - federatedfilesharing: 1.13.0
  - files: 1.18.0
  - files_mindmap: 0.0.26
  - files_pdfviewer: 2.4.0
  - files_rightclick: 1.2.0
  - files_sharing: 1.15.0
  - files_versions: 1.16.0
  - files_videoplayer: 1.12.0
  - lookup_server_connector: 1.11.0
  - maps: 0.2.1
  - music: 1.6.0
  - oauth2: 1.11.0
  - onlyoffice: 7.4.4
  - password_policy: 1.13.0
  - photos: 1.5.0
  - privacy: 1.7.0
  - provisioning_api: 1.13.0
  - ransomware_protection: 1.13.0
  - settings: 1.5.0
  - sharebymail: 1.13.0
  - survey_client: 1.11.0
  - text: 3.4.0
  - theming: 1.14.0
  - twofactor_backupcodes: 1.12.0
  - unsplash: 1.2.5
  - user_status: 1.3.1
  - viewer: 1.7.0
  - weather_status: 1.3.0
  - workflowengine: 2.5.0
```

**Comment by tacruc** [comment]
Am I correct, that what actually fails is deleting a category with many favorites? Or how do you delete the imported kml?

**Comment by mikeelucky** [comment]
06.09.2022, 15:24, "Arne Hamann" ***@***.***>: Am I correct, that what actually fails is deleting a category with many favorites? Or how do you delete the imported kml?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by siyuan-nz** [comment]
I'm not sure if what I observed is the same issue, but seems similar to me enough.

Steps to produce the problem for me:
1. Find a .gpx file (mine was exported from `PhoneTrack`)
2. Click on `...` of `My devices` in `Maps` -> `Import devices` -> Navigate to the .gpx file and import it
3. A new device appears under `My devices`, observe that no way can you find an option to delete the newly imported device, neither the `...` for `My devices` nor the `...` of the device listed under `My devices` offers an option to delete.

Basically I'm stuck with these imported devices with no obvious way to delete them, would be good if there's some command I could use to manually delete them before a UI option is implemented.

**Comment by beardhatcode** [comment]
![image](https://github.com/nextcloud/maps/assets/662538/44481a3a-d0b0-4f99-8df5-ea841a8b909b)

I'm also facing this issue, there is no option to delete a device that you imported

**Comment by siyuan-nz** [comment]
I don't know why, but the account of another user in my Nextcloud instance does show a `Delete all` option, but this option is missing from my own account.
![image](https://github.com/user-attachments/assets/b712e1fe-2557-4cb6-b834-80b22ec23c3d)
![image](https://github.com/user-attachments/assets/f53fbe2c-2ce5-4f3b-ac7b-e90e3d00b994)

**Comment by siyuan-nz** [comment]
Just found out with a test Nextcloud instance started fresh, installed Maps, then under `...` next to `My devices`, `Import devices`, select any valid .gpx file. Now that the device shows under `My devices`, there's no way to delete it, and the `Delete all` option from my previous comment disappears.

**Comment by mursonic** [comment]
I'm facing the same issue here... There is no way to delete imported devices in version 1.5.0

**Comment by mikeelucky** [comment]
It is unlikely that this error can be corrected without completely rewriting the source code of the program. When we try to delete marks, we pass an array of our points in the URL string, which clearly has a length limit, and this is where we encounter the problem with the impossibility of deleting multiple points.

**Comment by mursonic** [comment]
Just to get this straigt: There is a feature to add something, and no one thought it might be useful to remove the same thing?

**Comment by beardhatcode** [comment]
The php side has functions delete entire devices and entire tracks, so it is definitely doable. 

https://github.com/nextcloud/maps/blob/2bb9eabb7fbcbae2bf409e7fc130ecafe250da6b/lib/Service/TracksService.php#L477

**Comment by beardhatcode** [comment]
I was able to delete a device by using:
```js
 await fetch("/apps/maps/devices/DEVICE_ID_HERE", {
    "credentials": "include",
    "headers": {
        "Accept": "application/json, text/plain, */*",
        "Accept-Language": "en",
        "Content-Type": "application/json",
        "X-Requested-With": "XMLHttpRequest, XMLHttpRequest",
        "requesttoken": document.head.dataset.requesttoken,
        "Sec-GPC": "1",
        "Sec-Fetch-Dest": "empty",
        "Sec-Fetch-Mode": "cors",
        "Sec-Fetch-Site": "same-origin",
        "Pragma": "no-cache",
        "Cache-Control": "no-cache"
    },
    "method": "DELETE",
    "mode": "cors"
});
```

So the only this that needs to be done is add the right axios call I think
